src/check/constrain/generate/class.rs

```diff
@@ -24,13 +24,13 @@ pub fn gen_class(
         Node::Class { body: Some(body), ty, .. } | Node::TypeDef { body: Some(body), ty, .. } =>
             match &body.node {
                 Node::Block { statements } =>
-                    constrain_class_body(statements, ty, env, ctx, constr),
+                    constrain_class_body(statements, ty, env, ctx, constr, &ast.pos),
```

Comment:
Actually, this method probably shouldn't take a position.
Especially since each statement itself has a position, which could be used to create more local errors.
Now, if we have say, a very big class, then the error messages will refer to a large body of code and not be very helpful.